ros_tcp_endpoint/tcp_sender.py

```diff
@@ -125,7 +125,10 @@ def send_topic_list(self):
             topic_list = SysCommand_TopicsResponse()
             topics_and_types = self.tcp_server.get_topic_names_and_types()
             topic_list.topics = [item[0] for item in topics_and_types]
-            topic_list.types = [item[1] for item in topics_and_types]
+            if isinstance(topics_and_types[0][1], str):
+                topic_list.types = [item[1].replace("/msg/", "/") for item in topics_and_types]
+            else:
+                topic_list.types = [item[1][0].replace("/msg/", "/") for item in topics_and_types]
```

Comment:
Is the difference between ROS1/ROS2 syscommand format an intentional design choice/result of message generation differences, or is it a case of old format vs. new format?  If the latter, it'd be more maintainable to standardize on whatever the new format is for both 1 & 2, and print some kind of deprecation message here, prompting a user to upgrade their TCP-Connector, if we receive messages in the old format.

Reply:
it seems to be a ROS1 vs ROS2 difference, the format of what gets returned from [`get_topic_names_and_types().types`](https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Node.html#ab64ee81d61071100f222b819eab3e311) went from a list (`[sensor_msgs/Image, geometry_msgs/Pose]`) to a list of lists (`[[sensor_msgs/msg/Image], [geometry_msgs/msg/Pose]]`)

Reply:
Hmm, yikes. The intent is for them to be the same.

Reply:
Yeah so if that's the case then maybe we should declare this task blocked pending a bugfix that re-standardizes the format between 1 & 2 and prints the deprecation warning.